Submitted By:            Randy McMurchy <randy_at_linuxfromscratch_dot_org>
Date:                    2006-05-27
Initial Package Version: 2.2.11
Upstream Status:         In upstream CVS
Origin:                  Upstream CVS
Description:             Fixes an issue with Gimp using a deprecated libpng
                         function no longer available in >= libpng-1.2.9

$Id$


--- gimp-2.2.11-orig/plug-ins/common/png.c	2004-11-23 14:28:43.000000000 +0000
+++ gimp-2.2.11/plug-ins/common/png.c	2006-05-27 23:08:02.000000000 +0000
@@ -1012,7 +1012,7 @@
    * Done with the file...
    */
 
-  png_read_destroy (pp, info, NULL);
+  png_destroy_read_struct (&pp, &info, NULL);
 
   g_free (pixel);
   g_free (pixels);
@@ -1441,7 +1441,7 @@
     };
 
   png_write_end (pp, info);
-  png_write_destroy (pp);
+  png_destroy_write_struct (&pp, &info);
 
   g_free (pixel);
   g_free (pixels);
